File: Aula_2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main (void){
    string s = get_string("Text: \n");

    float letter = 0;
    float words = 1;
    float senteses = 0;

    for (int i = 0, n = strlen(s); i < n; i++){

        if(isalpha(s[i])){
            letter++;
        } else if (isspace(s[i])){
            words++;
        } else if (s[i] == '.' || s[i] == '?' || s[i] == '!'){
            senteses++;
        }
    }

    float L = letter / words * 100;
    float S = senteses / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int indexi = round(index);

    if(indexi >= 1 && indexi <= 16){
        printf("Grade %i\n", indexi);
    } else if (indexi < 1){
        printf("Before Grade 1\n");
    } else{
        printf("Grade 16+\n");
    }
}

File: Aula_6/Readability.py
import cs50

def main():
    text = cs50.get_string("Text: ")
    letters = 0
    senteces = 0
    words = 1


    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
        elif (text[i] == " "):
            words += 1
        elif (text[i] == "?" or text[i] == "!" or text[i] == "."):
            senteces += 1

    L = float (letters / words * 100) #average number of letters per 100 words
    S = float (senteces / words * 100) #average number of sentences per 100 words

    Grade = round((float)(0.0588 * L - 0.296 * S - 15.8))

    if Grade >= 16:
        print("Grade 16+")
    elif Grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {Grade}")

main()